### DISCUSSION TITLE
Strategies for supporting unreleased TOML features

### DISCUSSION THREAD

**Discussion by anhy999** [toml-lang/toml#923] [General]
### Discussed in https://github.com/toml-lang/toml/discussions/915

<div type='discussions-op-text'>

<sup>Originally posted by **awvwgk** July 31, 2022</sup>
What is the general strategy of TOML library implementors here to add support for unreleased TOML features?

Implementing features early before a release allows to provide access to users of TOML early, but comes with the disadvantage that the feature might change or other TOML related tooling is not compatible yet. I have seen some libraries provide a way to opt-in at compile time / run time for newer features.

Testing and validation is also a bit more tricky, especially when previously incorrect TOML documents are becoming valid.

Any thoughts on this are welcome.</div>

**Comment by marzer** [comment]
Not to be "that guy", but does this need to be an issue, if a discussion topic already exists? It's a pretty open-ended question; a discussion topic is probably already the right format.

**Comment by pradyunsg** [comment]
Please feel welcome to discuss this in #915.
